src/main/java/com/google/devtools/build/lib/rules/apple/DottedVersion.java

```diff
@@ -129,8 +129,11 @@ public static Option option(DottedVersion version) {
   private static final Splitter DOT_SPLITTER = Splitter.on('.');
   private static final Pattern COMPONENT_PATTERN =
       Pattern.compile("(\\d+)([a-z0-9]*?)?(\\d+)?", Pattern.CASE_INSENSITIVE);
+  private static final Pattern COMMENTIVE_COMPONENT_PATTERN =
+      Pattern.compile("([a-z]\\w*)", Pattern.CASE_INSENSITIVE);
```

Comment:
I'm wondering if we want be more restrictive here by making a list of common patterns, like "experimental", "testing", "internal". My concern is that we'll have some legit version that doesn't start with a number in one of the dotted components and things will break quietly.
If we know that the dotted version components will always start with a number, this is fine as is.

Reply:
AFAIK there is no version like that for now. This rule should work.